tq/ssh.go

```diff
@@ -73,6 +73,19 @@ func (a *SSHBatchClient) Batch(remote string, bReq *batchRequest) (*BatchRespons
 		return nil, fmt.Errorf("batch response: status %d from server (%s)", status, msg)
 	}
 
+	for _, arg := range args {
+		entries := strings.SplitN(arg, "=", 2)
+		if len(entries) < 2 {
+			continue
+		}
+		if entries[0] == "hash-algo" {
+			bRes.HashAlgorithm = entries[1]
+			if bRes.HashAlgorithm != "sha256" {
+				return nil, fmt.Errorf("batch response: unsupported hash algorithm: %q", entries[1])
+			}
+		}
+	}
+
```

Comment:
I may be missing something here, but I'm not clear on what this achieves, as I don't see how the `args` array would change from where it is set just above with the value `"hash-algo=sha256"`.  Is this code intended to future-proof against a time when the `args` array is set dynamically?  Or am I just missing how its contents are dynamic now?

Reply:
You're not missing anything.  I intended to make `batchInternal` return an `args` value, but didn't.  I've fixed that in the latest version.

Reply:
Okay, that makes more sense; it's parsing the arguments returned in the batch response.